lib/minfraud/minfraud_client.rb

```diff
@@ -69,13 +67,14 @@ def initialize(
 
     # @param url [String] URL endpoint, such as '/app/ccv2r'
     # @param transaction [Minfraud::Transaction] transaction to be sent to MaxMind
+    # @yield [Minfraud::Response] a block to pass to the response to allow for modification
     # @return [Minfraud::Response] wrapper for minFraud response
-    def send_transaction(transaction:, url: DEFAULT_API)
+    def send_transaction(transaction:, url: DEFAULT_API, &block)
       begin
         response = @http_client.get(url) do |req|
           req.params = self.class.encoded_query(transaction)
         end
-        Response.new(response)
+        Response.new(response, &block)
```

Comment:
Does a `Response` care about the block? Why not call the block from within this method?

Reply:
The block is passed in during initialization so it can be called before it freezes itself - this way, all Responses are guaranteed to be frozen, but it also allows for modification of the attributes during initialization

Added a comment to make this more clear